### PR TITLE
Draft of entitlements page

### PR DESCRIPTION
**Pull request by rasienko**
@sthielen @eltesoro-koji For your review, here is a new page that lists all of the entitlements in one place. It was a request from a community dev (and also part of the refactor of the deprecated packages, since that's where most of the entitlements were documented previously). Please let me know what you think and check for accuracy/completeness.Thanks!

### CODE REVIEW

**Review by eltesoro-koji — COMMENTED** [review]
Writing style suggestions:

1.
We should reconsider using the AsciiDoctor Button and Menu UI macros (https://docs.asciidoctor.org/asciidoc/latest/macros/ui-macros/). And maybe even create another macro/class for special terms so we can cut down on quotation marks. This separates the style from the content (kinda like the MVP model in programming and the reason the CSS is better separated from the HTML). It also makes our docs more like structured content because we're essentially tagging that piece of information. Even if the style is the same now, the macros/classes make it easier to change the style globally down the road.

2.
What do you think about descriptions for Boolean fields being in the form: "(_Boolean_) If `true`, .... [Otherwise, ....]"
Example: `InstantRemixing` – (_Boolean_) If `true`, creators can configure the app (`remix` context).
(I addressed this in some items below as examples.)

3.
Capitalization after a comma is not proper punctuation. I think we should place all data types inside parens.


In src/asciidoc/docs/platform-features/entitlements.adoc:

BEFORE:     To configure entitlements for your app, you define properties and values in the ...
SUGGESTION: To configure entitlements for your app, define properties and values in the ...

BEFORE: Your app can provide different experiences for different situations, or "contexts." To enable certain contexts, ...
BETTER: Your app can provide different experiences for different modes or contexts. To enable certain contexts, ...
BEST:   Your app can include other modes or contexts in addition to the default. To enable additional contexts, ...

BEFORE:     To enable financial transactions in the app (in-app purchases), configure the `InAppPurchases` object.
SUGGESTION: To enable financial transactions in the app, configure the `InAppPurchases` object.

BEFORE:  Enables in-app purchases in the app.
AFTER-OPTION 1: Enables in-app purchases.
AFTER-OPTION 2: Enables purchases in the app.

BEFORE: Price of the product in USD, for a product with a set price.
:  Price of the product in USD, if the product has a set price.
QUESTIONS:
  * Is it an integer or float?
  * If the price is not set, is the value set to 0 or undefined? Or is this key-value pair gone entirely?
  * @sthielen: If Europeans accidentally use a comma as the decimal point (followed by two digits), will the backend be able to handle it? Or will "1,00" be interpreted as "100" or "1,000"?

BEFORE:     Defines a product without a set price. For example, a donation or "pay what you want" price.
SUGGESTION: If `true`, the buyer can pay or donate any amount for the product. Used for pay-what-you-want (PWYW) pricing models.

CURRENT:  Enables the transaction receipt to display routes in the app.
QUESTION: What are "routes" in this sense?

BEFORE: Your can enable or disable some platform navigation features for your app.
AFTER:  You can enable or disable some platform navigation features for your app.

BEFORE:     If your app <<_contexts, allows creators to configure it>>, you can control the navigation for the configuration experience.
SUGGESTION: If your app is <<_contexts#,configurable>>, you can control the navigation for the configuration experience.

BEFORE:     Hides the platform action for advancing from configuration to preview, so that the app can control this action.
SUGGESTION: If `true`, the user cannot transition from the configuration context to the preview context; the app controls this action.

BEFORE:     Hides the platform action for cancelling a remix, so that the app can control this action.
SUGGESTION: If `true`, the user cannot cancel a customization session; the app controls this action.

**Comment by rasienko** [comment]
Thanks @eltesoro-koji! Clarified per your suggestions. To answer your questions...

1. It might be a good time to consider the UI macros, since we have to update all of the navigation re: the recent platform changes. Let’s discuss as a separate ticket. 
2. Re: the formatting, the TypeDoc parser already uses this format, so we need to follow it for now. We can consider a change if/when we update the parser.

@sthielen, what is the type of the product price (`InAppPurchases.products[index].price`)? Also, let me know if there are any other inaccuracies or changes, or if we're ready to publish this page. Thanks!

File: src/asciidoc/docs/platform-features/entitlements.adoc [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+<2> `AboutContext` – _Boolean_, Enables a distinct experience when a creator first configures the app (`about` context).
+Default is `false`.
++
+NOTE: If this entitlement is enabled, the app must use <<core-frontend-ui-navigate#createRemix, Koji.ui.navigate.createRemix>> to open the configuration options from the About page.
+
+<3> `AdminContext` – _Boolean_, Enables a distinct experience for the creator (`admin` context).
+For example, fulfillment for product sales apps or content moderation for social post apps.
+If enabled, the creator can access the admin page by going to *App > ... > Admin Page*.
+Default is `false`.
+<4> `StickerContext` – _Boolean_, Enables a distinct experience when the app is embedded in another Koji app (`sticker` context).
```
[the remark this answers is not in the document]
Comment (rasienko) [reply src/asciidoc/docs/platform-features/entitlements.adoc]: Removed from this page and throughout the docs. Will add a deprecation notice to the next news update.

File: src/asciidoc/docs/platform-features/entitlements.adoc [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    "AboutContext": true, <2>
+    "AdminContext": true, <3>
+    "StickerContext": true <4>
+  }
+}
+----
+<1> `InstantRemixing` – _Boolean_, Allows creators to configure the app (`remix` context).
+Default is `false`.
+Optionally, configure entitlements to <<_navigation, control the navigation>> for the configuration experience.
+<2> `AboutContext` – _Boolean_, Enables a distinct experience when a creator first configures the app (`about` context).
```
[the remark this answers is not in the document]
Comment (rasienko) [reply src/asciidoc/docs/platform-features/entitlements.adoc]: Removed from this page and throughout the docs. Will add a deprecation notice to the next news update.

File: src/asciidoc/docs/platform-features/entitlements.adoc [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+Default is `false`.
++
+NOTE: If this entitlement is enabled, the app must use <<core-frontend-ui-navigate#createRemix, Koji.ui.navigate.createRemix>> to open the configuration options from the About page.
+
+<3> `AdminContext` – _Boolean_, Enables a distinct experience for the creator (`admin` context).
```
[the remark this answers is not in the document]
Comment (rasienko) [reply src/asciidoc/docs/platform-features/entitlements.adoc]: Removed from this page and throughout the docs. Will add a deprecation notice to the next news update.

File: src/asciidoc/docs/platform-features/entitlements.adoc [hunk 4]
```diff
@@ -12,39 +12,37 @@ When you initialize the @withkoji/core package with this configuration data, the
...
-If enabled, the creator can access the admin page by going to *App > ... > Admin Page*.
-Default is `false`.
-<4> `StickerContext` – _Boolean_, Enables a distinct experience when the app is embedded in another Koji app (`sticker` context).
+<3> `InstantRemixingNativeCancel` – _Boolean_, Hides the platform action for cancelling a configuration, so that the app can provide this action.
 Default is `false`.
++
+NOTE: If this entitlement is enabled, the app must use <<core-frontend-remix#cancel, Koji.remix.cancel>> to implement a custom cancel action.
 
-NOTE: To use the `receipt` context, you must <<_in_app_purchases, configure in-app purchases>> and enable dynamic receipts.
+<4> `FeedEvents` – _Boolean_, Enables the app to be displayed in a platform feed.
```
Comment (sthielen) [inline src/asciidoc/docs/platform-features/entitlements.adoc:45]: This one is also deprecated! Sorry for not catching on the first pass 😅

File: src/asciidoc/reference/deprecated/withkojiVCC.adoc [hunk 5]
```diff
@@ -474,14 +437,14 @@ Enables distinct template experiences for different viewing contexts.
 
 * `about` – Manages the context for the remix landing page (About page).
 To use this context:
-** *Define an About page experience* – Enable the `AboutContext` <<_feedsdk, entitlement>>, and define a view intended for the About page.
+** *Define an About page experience* – Enable the `AboutContext` entitlement, and define a view intended for the About page.
```
Comment (sthielen) [inline src/asciidoc/reference/deprecated/withkojiVCC.adoc:440]: Are these and below still supposed to be here?

Comment (rasienko) [reply src/asciidoc/reference/deprecated/withkojiVCC.adoc:440]: Nah, we don't need these deprecated instructions. Removed and just left the generic bullet (for now, until we remove those packages from the site entirely).

File: src/asciidoc/docs/platform-features/entitlements.adoc (hunk 4 above)
Comment (rasienko) [reply src/asciidoc/docs/platform-features/entitlements.adoc:45]: Thanks! Removed this one throughout, as well.

**Review by sthielen — APPROVED** [review]

**Pull request merged**